js/apps/admin-ui/src/util.ts

```diff
@@ -171,3 +171,18 @@ export const addTrailingSlash = (url: string) =>
   url.endsWith("/") ? url : url + "/";
 
 export const generateId = () => Math.floor(Math.random() * 1000);
+
+type PathParam = { [key: string]: string | null };
+
+export function generateUnencodedPath<Path extends string>(
```

Comment:
Since this function generates a path that is URI encoded, should this not be called `generateEncodedPath()` instead?
```suggestion
export function generateEncodedPath<Path extends string>(
```